### ISSUE TITLE
CreateIndex() not working?

### ISSUE THREAD

**Issue by ngbrown** [elastic/elasticsearch-net#2445]
**NEST/Elasticsearch.Net version**:
I'm using NEST v5.0.0-rc4 with Elasticsearch v5.0.2.

Based on the documentation, this should work:

```cs
[ElasticsearchType(Name = "user")]
public class SearchUser
{
    public string Username { get; set; }

    [Keyword]
    public string[] Is { get; set; }
}

var descriptor = new CreateIndexDescriptor("t2007")
    .Mappings(ms => ms
        .Map<SearchUser>(m => m.AutoMap())
    );
client.CreateIndex(descriptor);
```

It creates the index, but the resulting `_mapping` in Elasticsearch is empty.  Indeed, the request looks like this:
```
PUT http://localhost:9200/t2007?pretty=true 
{}
```

Am I missing something in the documentation or is there something majorly wrong with the client?

**Comment by ngbrown** [comment]
I can manually map my example fine in Kibana dev-tools like this:

```json
PUT /tis2007
{
  "mappings": {
    "user": {
      "properties": {
        "username": {
          "type": "text"
        },
        "is": {
          "type": "keyword"
        }
      }
    }
  }
}
```

My actual entities are quite a bit larger and I was hoping to use the auto-map feature.

**Comment by russcam** [comment]
I can't replicate this with 5.0.0-rc4; is it possible that the index is already created when trying to create it - in this case, the mapping won't be applied as the create index request will result in a 400.

Here's an example

```c#
void Main()
{
    var host = Process.GetProcessesByName("fiddler").Any() 
        ? "ipv4.fiddler" 
        : "localhost";
    
    var pool = new SingleNodeConnectionPool(new Uri($"http://{host}:9200"));
    var defaultIndex = "default-index";
    var connectionSettings = new ConnectionSettings(pool)
            .DefaultIndex(defaultIndex)
            .PrettyJson()
            .DisableDirectStreaming()
            .OnRequestCompleted(response =>
                {
                    // log out the request
                    if (response.RequestBodyInBytes != null)
                    {
                        Console.WriteLine(
                            $"{response.HttpMethod} {response.Uri} \n" +
                            $"{Encoding.UTF8.GetString(response.RequestBodyInBytes)}");
                    }
                    else
                    {
                        Console.WriteLine($"{response.HttpMethod} {response.Uri}");
                    }

                    Console.WriteLine();

                    // log out the response
                    if (response.ResponseBodyInBytes != null)
                    {
                        Console.WriteLine($"Status: {response.HttpStatusCode}\n" +
                                 $"BODY:{Encoding.UTF8.GetString(response.ResponseBodyInBytes)}\n" +
                                 $"{new string('-', 30)}\n");
                    }
                    else
                    {
                        Console.WriteLine($"Status: {response.HttpStatusCode}\n" +
                                 $"{new string('-', 30)}\n");
                    }
                });

    var client = new ElasticClient(connectionSettings);

    var descriptor = new CreateIndexDescriptor("t2007")
        .Mappings(ms => ms
            .Map<SearchUser>(m => m.AutoMap())
        );
    
    
    var createIndexResponse = client.CreateIndex(descriptor);
    	
    if (!createIndexResponse.IsValid)
        throw new Exception(createIndexResponse.DebugInformation);
        
    client.GetMapping<SearchUser>(m => m.Index("t2007"));

}

[ElasticsearchType(Name = "user")]
public class SearchUser
{
    public string Username { get; set; }

    [Keyword]
    public string[] Is { get; set; }
}
```

Running this in Linqpad against a running local Elasticsearch 5.x instance, we see the following in the console output

```json
PUT http://localhost:9200/t2007?pretty=true 
{
  "mappings": {
    "user": {
      "properties": {
        "username": {
          "fields": {
            "keyword": {
              "ignore_above": 256,
              "type": "keyword"
            }
          },
          "type": "text"
        },
        "is": {
          "type": "keyword"
        }
      }
    }
  }
}

Status: 200
BODY:{
  "acknowledged" : true,
  "shards_acknowledged" : true
}

------------------------------

GET http://localhost:9200/t2007/_mapping/user?pretty=true

Status: 200
BODY:{
  "t2007" : {
    "mappings" : {
      "user" : {
        "properties" : {
          "is" : {
            "type" : "keyword"
          },
          "username" : {
            "type" : "text",
            "fields" : {
              "keyword" : {
                "type" : "keyword",
                "ignore_above" : 256
              }
            }
          }
        }
      }
    }
  }
}

------------------------------
```

If you try running it again, you'll see the exception thrown

```json
PUT http://localhost:9200/t2007?pretty=true 
{
  "mappings": {
    "user": {
      "properties": {
        "username": {
          "fields": {
            "keyword": {
              "ignore_above": 256,
              "type": "keyword"
            }
          },
          "type": "text"
        },
        "is": {
          "type": "keyword"
        }
      }
    }
  }
}

Status: 400
BODY:{
  "error" : {
    "root_cause" : [
      {
        "type" : "index_already_exists_exception",
        "reason" : "index [t2007/JRcJDGq_Qnmpowh7T2toLA] already exists",
        "index_uuid" : "JRcJDGq_Qnmpowh7T2toLA",
        "index" : "t2007"
      }
    ],
    "type" : "index_already_exists_exception",
    "reason" : "index [t2007/JRcJDGq_Qnmpowh7T2toLA] already exists",
    "index_uuid" : "JRcJDGq_Qnmpowh7T2toLA",
    "index" : "t2007"
  },
  "status" : 400
}

------------------------------
```

**Comment by russcam** [comment]
If the index is already created and doesn't contain the mapping for `SearchUser`, you can use the `.Map<T>()` (or the async variant) to add the mapping; this is exactly the same as what you're doing in Kibana console.

Hope that helps!

**Comment by ngbrown** [comment]
Ok, I left out all the setup, but it was very close to what you had, I was deleting the index before and watching the request and response through the `list` (from the connection example in the document).

The major change, which seems to matter that I had this:

```cs
var settings = new ConnectionSettings(connectionPool, con => new JsonNetSerializer(con, (s, values) => s.ContractResolver = new CamelCasePropertyNamesContractResolver()))
    .PrettyJson(true)
    .DisableDirectStreaming()
    .OnRequestCompleted(response => {...similar to the stuff you had...});
```

I was originally using the low level client and wanted to de-capitalize the field names using Json.Net so I thought I had to adjust the serializer with the `CamelCasePropertyNamesContractResolver` like I usually do when using Json.Net.

By removing the second parameter on `ConnectionSettings`, it is working correctly.  What did I do wrong with using `JsonNetSerializer`?

**Comment by russcam** [comment]
Unfortunately, NEST sets its own `IContractResolver` for the JsonNetSerializer that is needed for a number of reasons, one of which is to carry over state in certain scenarios that is needed at deserialization time. Because of this, if you override the `ContractResolver`, be sure to derive from [`ElasticContractResolver`](https://github.com/elastic/elasticsearch-net/blob/dde789b6adcd8104f44d23e9cc3d6869a969e648/src/Nest/CommonAbstractions/SerializationBehavior/ElasticContractResolver.cs).

NEST camel cases all properties by default, so you probably don't need to set your own resolver for this; `ConnectionSettings` has a method,[ `DefaultFieldNameInferrer()`](https://www.elastic.co/guide/en/elasticsearch/client/net-api/5.x/field-inference.html#camel-casing), that can be passed a `Func<string, string>` to determine how field names are serialized.

**Comment by ngbrown** [comment]
Ok.  I see that it is doing camel casing, so the default behavior is what I wanted.

I just made a bad assumption because the low-level client didn't do camel casing and it looked like it would be hard to change it, so that's part of why I moved to NEST, because I knew how to override that with JSON.Net.  I was mystified by the problem as there were no exceptions or warnings or anything.

Thanks for your help.
